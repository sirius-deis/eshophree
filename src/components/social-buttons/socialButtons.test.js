import { render, screen, fireEvent } from "@testing-library/react";
import SocialButtons from "./socialButtons";


describe("SocialButtons component", () => {
  const socialButtons = [{button: <div>Google</div>, onClick: () => {}}, {button: <div>Facebook</div>, onClick: () => {}}];
  it("should match snapshot", () => {
    const { container } = render(<SocialButtons title="login" buttons={socialButtons} />);
    expect(container).toMatchSnapshot();
  });

  it("should render component with given title", () => {
    render(<SocialButtons title="signup" buttons={socialButtons} />);
    expect(screen.getByText(/signup/)).toBeInTheDocument();
  });

  it("should render component with given social buttons", () => {
    render(<SocialButtons title="signup" buttons={socialButtons} />);
    expect(screen.getByText(/google/i)).toBeInTheDocument();
    expect(screen.getByText(/facebook/i)).toBeInTheDocument();
  });
});