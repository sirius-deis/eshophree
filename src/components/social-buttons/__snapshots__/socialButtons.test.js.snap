// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialButtons component should match snapshot 1`] = `
<div>
  <div
    class="sc-iVCKna eKaoQc"
  >
    <div
      class="sc-gxYJeL fIRewz"
    >
      <div
        class="sc-bgqQcB teJYB"
        width="35"
      />
      <p
        class="sc-kPTPQs iPJAVP"
      >
        Or 
        login
         with
      </p>
      <div
        class="sc-bgqQcB teJYB"
        width="35"
      />
    </div>
    <div
      class="sc-iVCKna eKaoQc"
    >
      <button
        class="sc-gTRrQi sc-dicizt isQxaZ dkhXdi"
        type="button"
      >
        <div>
          Google
        </div>
      </button>
      <button
        class="sc-gTRrQi sc-dicizt isQxaZ dkhXdi"
        type="button"
      >
        <div>
          Facebook
        </div>
      </button>
    </div>
  </div>
</div>
`;
